src/library.js

```diff
@@ -199,8 +203,9 @@ mergeInto(LibraryManager.library, {
   emscripten_resize_heap: 'ip',
   emscripten_resize_heap: (requestedSize) => {
     var oldSize = HEAPU8.length;
-#if MEMORY64 != 1
-    requestedSize = requestedSize >>> 0;
+#if !MEMORY64 && !CAN_ADDRESS_2GB
+    // With CAN_ADDRESS_2GB or MEMORY64, pointers are already unsigned.
+    requestedSize >>>= 0;
```

Comment:
Why do this? The code is only run if `requestedSize === (requestedSize >>> 0)`.

Reply:
The reason for this special case is that we still want `emscripten_resize_heap(MAX_INT32)` to fail here even if `CAN_ADDRESS_2GB` is not set.. at least the tests expect that.  So this function in particular has to force its parameter to unsigned even when `CAN_ADDRESS_2GB` is not set.

Reply:
In fact we want to fail *especially* if `CAN_ADDRESS_2GB` is not set.  

Without this line the below `assert(requestedSize > oldSize);` will fail because requestedSize would come through as `-1`.